libs/feature/map/src/lib/utils/map-utils.service.ts

```diff
@@ -104,4 +109,34 @@ export class MapUtilsService {
       return []
     }
   }
+
+  getLayerExtent(layer: MapContextLayerModel): Observable<Extent> {
+    let extent = []
+    if (layer.type === 'geojson' || layer.type === 'wfs') {
+      const features = new GeoJSON().readFeatures(layer.data)
+      if (!features[0]) return of(undefined)
+      extent = features[0].getGeometry().getExtent()
+      features.forEach((feature) => {
+        extent = extend(extent, feature.getGeometry().getExtent())
+      })
+      extent = this.extentFromLonLat(extent)
+    } else if (layer.type === 'wms') {
+      return from(
+        new WmsEndpoint(layer.url).isReady().then((endpoint) => {
+          const wmsLayer = endpoint.getLayerByName(layer.name)
+          return this.extentFromLonLat(wmsLayer.boundingBoxes['EPSG:4326'])
+        })
+      ) as Observable<Extent>
+    } else {
+      return of(undefined)
+    }
+    return of(extent)
```

Comment:
This is not needed (will never be executed)

Reply:
ok, shouldn't we keep a fall back for the case once we might have other types than `wms` and `geojson`?

Reply:
we definitely need a fallback :) it's good now, thanks